pkg/slo-controller/noderesource/plugins/resourceamplification/plugin.go

```diff
@@ -111,3 +125,50 @@ func (p *Plugin) Calculate(_ *configuration.ColocationStrategy, node *corev1.Nod
 		},
 	}, nil
 }
+
+func calculateCPUAmplificationRatio(node *corev1.Node, ratio extension.Ratio) extension.Ratio {
+	var finalRatio extension.Ratio = 1
+	if ratio > 0 {
+		finalRatio = ratio
+	}
+
+	// do not return error, print warning log and use config amplification ratio
+	cpuNormalizationRatio, err := extension.GetCPUNormalizationRatio(node)
+	if err != nil {
+		klog.Warningf("failed to get cpu normalization ratio: %v, directly use cpu amplification ratio if it has value", err)
+		return finalRatio
+	}
+
+	if cpuNormalizationRatio > 0 {
+		finalRatio *= extension.Ratio(cpuNormalizationRatio)
+	}
+
+	return finalRatio
+}
+
+func isAmplificationRatioValid(ratios map[corev1.ResourceName]extension.Ratio) error {
+	if ratios == nil {
+		return fmt.Errorf("ratios is nil")
+	}
+
+	for k, v := range ratios {
+		if v < 1 {
+			return fmt.Errorf("ratio with resource type %v and value %v is invalid", k, v)
+		}
+	}
+
+	return nil
+}
+
+func getResourceAmplificationRatio(node *corev1.Node) map[corev1.ResourceName]extension.Ratio {
+	ratios := map[corev1.ResourceName]extension.Ratio{}
+	if cfgHandler != nil && cfgHandler.IsCfgAvailable() {
```

Comment:
No need to judge nil here, but if the config is not available, we should return error rather than ignore.

Reply:
> No need to judge nil here, but if the config is not available, we should return error rather than ignore.

It may be more general. In the calculate process, it does not need to check the pointer.
I will change the logic, it does not judge nil, and return error when `IsCfgAvailable=false` and `GetCPUNormalizationRatio=false`